src/mongoose_wpool.erl

```diff
@@ -0,0 +1,42 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% This is here because there are pool options which have to be given when calling
+%%% the pool (selection strategy, timeout), while we want to set it once for the pool and not
+%%% worry about them later, hence additional storage.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_wpool).
+-author("bartlomiej.gorny@erlang-solutions.com").
+-include("mongoose.hrl").
+%% API
+-export([setup_env/0, get_pool_settings/1, delete_pool_settings/1, save_pool_settings/2]).
+
+setup_env() ->
```

Comment:
I see this function called in different places where pools are needed. Since we unified pools in this PR and we always use `worker_pool` for all pools I suggest to call this function only once, somewhere in `ejabberd_app`.

Reply:
It'd be even more compliant with OTP principles if `wpool` application where started automatically when MongooseIM is starting.

Reply:
Ok, I will start it from `ejabberd_app`, but would you mind if I leave the rest of the call, at least for now? It ensures everything is set up, somewhat like `mongoose_metrics:ensure_metric/3`, and as such is very handy for running small tests. Removing those calls would require fixing a couple of small tests, for which I may not have time today.

Reply:
I think I'll rename it to `ensure_started`

Reply:
Ok, that's fine.